src/token/from-request.js

```diff
@@ -0,0 +1,36 @@
+import Debug from 'debug';
+
+const debug = Debug('feathers-authentication:token:from-request');
+
+export default function(options) {
+  const header = options.header;
+
+  if (!header) {
+    throw new Error(`'header' property must be set in authentication options`);
+  }
+
+  return function fromRequest(data) {
+    if (typeof data === 'object' && data.headers) {
+      const req = data;
+
+      debug('Parsing token from request');
+
+      // Normalize header capitalization the same way Node.js does
+      let token = req.headers && req.headers[header.toLowerCase()];
```

Comment:
Do we want to support body and query string as well or just header? If it's not going to cause us problems maybe we should (at least body). I personally use querystring a lot in dev mode because I'm lazy but really we shouldn't support that officially.